### DISCUSSION TITLE
Image Test throws error message

### DISCUSSION THREAD

**Discussion by liebeckm** [neuefische/questions#549] [Web] [answered]
Hey,

I am writing tests right now. At my overview-page, there are images displayed. Therefore, I wanted to test that.
The test has the following error message: 

"TestingLibraryElementError: Unable to find an accessible element with the role "image""

I already tried getByRole, getById, getByAltText; I also changed the import from "Image" to "ListOfRides"...
Maybe it has something to do with Rendering?

Can someone tell me how to write a correct test for testing images?

please have a look at
pages/rides-overview/index.js
and
pages/rides-overview/index.test.js

Here is my code:
https://github.com/liebeckm/capstone-project/tree/list-of-rides

Thank you so much!

**Comment by PReichetanz** [comment, marked as answer]
Hi Melissa,

the main problem was the import and rendering of the `ListOfRides` component into your test file: the component is a default export, so we don't need curly braces `{}`, and we pass it to the `render()` function as JSX (with `<>`).

```js
import ListofRides from '.';
import { render, screen } from '@testing-library/react';

test('Image has alt text', () => {
  render(<ListofRides />);
  const element = screen.getAllByRole('img');
  expect(element[0]).toHaveAttribute('alt', 'profile pic');
});

```

The image role is called "img" and to prevent an error that there are multiple elements with this role, I changed the query to `getAllByRole` and check only for the first element.

This is not perfect, but should at least make sure the first element has an `alt` attribute.

What you need to change as well is the `key` for the list:

```js
// pages/rides-overview/index.js
<StyledCard key={ride.id}>
```

It would be better to move the List into its own component and pass it the `/lib/rides` as a prop. This way, you could better test all images which are rendered by mocking the prop itself.
